### PR TITLE
[Merged by Bors] - feat: `∃ᶠ x in 𝓝 a, x < a`

### PR DESCRIPTION
**Pull request by urkud**
* Add `frequently_lt_nhds` and `frequently_gt_nhds`.
* Move some lemmas from `Topology/Order/Basic` to `Topology/Algebra/Order/LeftRight`.
* Relax TC assumptions in `Filter.Eventually.exists_lt` (and `*_gt`).
  New versions assume `NeBot (𝓝[<] a)` and `NeBot (𝓝[>] a)`,
  so the latter one can be applied, e.g., to `((a : ℝ≥0) : ℝ≥0∞)`.

From the Mandelbrot set connectedness project.

Co-Authored-By: @girving 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Algebra/Order/LeftRight.lean [hunk 1]
```diff
@@ -27,6 +27,46 @@ left continuous, right continuous
...
+theorem Filter.Eventually.exists_gt {a : α} [NeBot (𝓝[>] a)] {p : α → Prop}
+    (h : ∀ᶠ x in 𝓝 a, p x) : ∃ b > a, p b :=
+  ((frequently_gt_nhds a).and_eventually h).exists
+#align filter.eventually.exists_gt Filter.Eventually.exists_gt
+
+theorem nhdsWithin_Ici_neBot {a b : α} (H₂ : a ≤ b) : NeBot (𝓝[Ici a] b) :=
+  nhdsWithin_neBot_of_mem H₂
+#align nhds_within_Ici_ne_bot nhdsWithin_Ici_neBot
+
+@[instance]
```
Comment (sgouezel) [inline Mathlib/Topology/Algebra/Order/LeftRight.lean:54]: Do you understand why this a `theorem` with `@[instance]` instead of just `instance`?

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

File: Mathlib/Topology/Algebra/Order/LeftRight.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Topology/Algebra/Order/LeftRight.lean:54]: It looks like I did it at the same time I changed it to use `ne_bot` in Lean 3 times. I don't remember why.

File: Mathlib/Topology/Algebra/Order/LeftRight.lean [hunk 2]
```diff
@@ -27,6 +27,46 @@ left continuous, right continuous
...
+theorem nhdsWithin_Ici_self_neBot (a : α) : NeBot (𝓝[≥] a) :=
```
Comment (urkud) [inline Mathlib/Topology/Algebra/Order/LeftRight.lean:55]:
```suggestion
instance nhdsWithin_Ici_self_neBot (a : α) : NeBot (𝓝[≥] a) :=
```

File: Mathlib/Topology/Algebra/Order/LeftRight.lean [hunk 3]
```diff
@@ -27,6 +27,46 @@ left continuous, right continuous
...
+  nhdsWithin_Ici_neBot (le_refl a)
+#align nhds_within_Ici_self_ne_bot nhdsWithin_Ici_self_neBot
+
+theorem nhdsWithin_Iic_neBot {a b : α} (H : a ≤ b) : NeBot (𝓝[Iic b] a) :=
+  nhdsWithin_neBot_of_mem H
+#align nhds_within_Iic_ne_bot nhdsWithin_Iic_neBot
+
+@[instance]
+theorem nhdsWithin_Iic_self_neBot (a : α) : NeBot (𝓝[≤] a) :=
```
Comment (urkud) [inline Mathlib/Topology/Algebra/Order/LeftRight.lean:64]:
```suggestion
instance nhdsWithin_Iic_self_neBot (a : α) : NeBot (𝓝[≤] a) :=
```

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**